### PR TITLE
prepare the contents of the npm package

### LINKED ISSUE #16: [components] publish the source code (tsx) in the package

### PR DESCRIPTION
**Pull request by severo**
fixes #16 

- tsc populates the new es/ directory with transpiled files (ts/tsx -> js + d.ts)
- vite generates minified bundled files (es and umd) in dist/, with their source maps

package.json should point to the appropriate files in the npm package.

Contents of the npm package

```
$ npm publish --dry-run

> @hyparam/components@0.1.0 prepublishOnly
> npm run build

> @hyparam/components@0.1.0 build
> tsc && vite build

vite v5.4.10 building for production...
✓ 42 modules transformed.
dist/index.es.min.js  236.69 kB │ gzip: 123.91 kB │ map: 126.18 kB dist/index.umd.min.js  225.15 kB │ gzip: 122.52 kB │ map: 122.83 kB ✓ built in 747ms
npm notice
npm notice 📦  @hyparam/components@0.1.0
npm notice Tarball Contents
npm notice 1.0kB LICENSE
npm notice 101B README.md
npm notice 408.4kB dist/assets/parquetWorker-zWrM-10F.js.map
npm notice 236.7kB dist/index.es.min.js
npm notice 126.2kB dist/index.es.min.js.map
npm notice 225.2kB dist/index.umd.min.js
npm notice 122.8kB dist/index.umd.min.js.map
npm notice 72B es/components/App.d.ts
npm notice 142B es/components/App.js
npm notice 265B es/components/Breadcrumb.d.ts
npm notice 989B es/components/Breadcrumb.js
npm notice 322B es/components/Cell.d.ts
npm notice 3.5kB es/components/Cell.js
npm notice 242B es/components/File.d.ts
npm notice 607B es/components/File.js
npm notice 243B es/components/Folder.d.ts
npm notice 2.3kB es/components/Folder.js
npm notice 416B es/components/index.d.ts
npm notice 416B es/components/index.js
npm notice 824B es/components/Layout.d.ts
npm notice 1.6kB es/components/Layout.js
npm notice 197B es/components/Markdown.d.ts
npm notice 3.7kB es/components/Markdown.js
npm notice 73B es/components/Page.d.ts
npm notice 873B es/components/Page.js
npm notice 428B es/components/Viewer.d.ts
npm notice 1.0kB es/components/Viewer.js
npm notice 416B es/components/viewers/ContentHeader.d.ts
npm notice 456B es/components/viewers/ContentHeader.js
npm notice 321B es/components/viewers/ImageView.d.ts
npm notice 2.8kB es/components/viewers/ImageView.js
npm notice 290B es/components/viewers/index.d.ts
npm notice 290B es/components/viewers/index.js
npm notice 327B es/components/viewers/MarkdownView.d.ts
npm notice 2.1kB es/components/viewers/MarkdownView.js
npm notice 389B es/components/viewers/ParquetView.d.ts
npm notice 3.0kB es/components/viewers/ParquetView.js
npm notice 376B es/components/viewers/TextView.d.ts
npm notice 2.4kB es/components/viewers/TextView.js
npm notice 1.7kB es/lib/files.d.ts
npm notice 2.7kB es/lib/files.js
npm notice 432B es/lib/key.d.ts
npm notice 698B es/lib/key.js
npm notice 301B es/lib/tableProvider.d.ts
npm notice 4.2kB es/lib/tableProvider.js
npm notice 125B es/lib/utils.d.ts
npm notice 125B es/lib/utils.js
npm notice 11B es/workers/parquetWorker.d.ts
npm notice 2.7kB es/workers/parquetWorker.js
npm notice 641B es/workers/parquetWorkerClient.d.ts
npm notice 2.8kB es/workers/parquetWorkerClient.js
npm notice 1.9kB package.json
npm notice Tarball Details
npm notice name: @hyparam/components
npm notice version: 0.1.0
npm notice filename: hyparam-components-0.1.0.tgz
npm notice package size: 472.3 kB
npm notice unpacked size: 1.2 MB
npm notice shasum: 41eba030a842e4fc1b13f8990d5accba4d1cf431
npm notice integrity: sha512-89CuWNs2a/I8z[...]4Jk4kJmwUb8lA==
npm notice total files: 52
npm notice
npm warn This command requires you to be logged in to https://registry.npmjs.org/ (dry-run)
npm notice Publishing to https://registry.npmjs.org/ with tag latest and default access (dry-run)
+ @hyparam/components@0.1.0
```

### CODE REVIEW

File: packages/components/package.json [hunk 1]
```diff
@@ -15,18 +15,26 @@
...
-      "import": "./dist/index.es.js",
-      "require": "./dist/index.umd.js",
-      "types": "./dist/components/index.d.ts"
-    }
+      "import": "./es/components/index.js",
+      "require": "./dist/index.umd.min.js",
+      "types": "./es/components/index.d.ts"
+    },
+    "./components/*": "./es/components/*.js",
+    "./components/*.js": "./es/components/*.js",
```
Comment (platypii) [inline packages/components/package.json:29]: I don't understand these lines. Why exporting `*` and `*.js` as `*.js`?

**Review by platypii — APPROVED** [review]
One question but overall looks good.

**Comment by severo** [comment]
I'll double check but I understand that it let's the library client import with both syntaxes:
import ... from "lib/files"
import ... from "lib/files.js"

Maybe not important though, and better to provide only one

-------- Message d'origine --------

**Comment by platypii** [comment]
I'd prefer just one, if both are not needed. Can always add more aliases later

File: packages/components/package.json (hunk 1 above)
Comment (severo) [reply packages/components/package.json]: fixed

**Pull request merged**